src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+// @ts-nocheck
+/* eslint-disable func-names */
+/* global fetch, process */
+
+/** 
+ * IMPORTS
+ */
+import { execSwingsetTransaction } from '../lib/chain.js';
+import {
+  makeRpcUtils,
+  boardSlottingMarshaller,
+  networkConfig,
+  makeAgoricNames
+} from '../lib/rpc.js'
+import axios from 'axios';
+import http from 'http';
+import fs from 'fs';
+import { getCurrent } from '../lib/wallet.js';
+import bodyParser from 'body-parser';
+import express from 'express';
+import { validUrl, delay, readJSONFile, readVStorage } from '../helpers/utils.js';
+import {
+  makeFollower,
+  makeLeader,
+} from '@agoric/casting';
+import { iterateReverse } from '@agoric/casting';
+import { getAllJobs, createDBs, createJob, deleteJob, queryTable, updateTable} from '../helpers/db.js'
+
+// get environment variables
+const {
+  PORT = '3000',
+  EI_CHAINLINKURL,
+  FROM,
+  SUBMIT_RETRIES = 3,
+  BLOCK_INTERVAL = '6',
+  SEND_CHECK_INTERVAL = '45',
+  AGORIC_RPC = "http://0.0.0.0:26657",
+  STATE_FILE = "data/middleware_state.json",
+  CREDENTIALS_FILE = "config/ei_credentials.json",
+  OFFERS_FILE = "config/offers.json",
+  FEEDS_FILE = "oracle/feeds.json"
+} = process.env;
+
+/** 
+  * Environment variables validation
+  */
+if (process.env.NODE_ENV != "test"){
+  assert(EI_CHAINLINKURL, '$EI_CHAINLINKURL is required');
+  assert(Number(SUBMIT_RETRIES), '$SUBMIT_RETRIES is required');
+  assert(FROM, '$FROM is required');
+  assert(validUrl(AGORIC_RPC), '$AGORIC_RPC is required');
+  assert(STATE_FILE != "", '$STATE_FILE is required');
+  assert(CREDENTIALS_FILE != "", '$CREDENTIALS_FILE is required');
+  assert(OFFERS_FILE != "", '$OFFERS_FILE is required');
+  assert(FEEDS_FILE != "", '$FEEDS_FILE is required');
```

Comment:
What's the point of checking if environment variables are empty if you're pre-assigning them just above?

Reply:
For files I would just if the file exists

Reply:
Done in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/78b48d44d89642075ce5573da789130283de6bf5